src/requests/cache/set_add_elements.rs

```diff
@@ -42,6 +43,11 @@ impl<S: IntoBytes, E: IntoBytes> SetAddElementsRequest<S, E> {
             collection_ttl: Some(collection_ttl),
         }
     }
+
+    pub fn with_ttl(mut self, collection_ttl: CollectionTtl) -> Self {
```

Comment:
thanks for adding this.

do you have any strong opinions on how we name these going forward? I was waffling between using the `with_` prefix or not. I might be leaning towards not using that prefix, because in some stuff I've worked with in the past, it was used to signify a copy constructor (thus, leaving the original object intact without mutating it, whereas this is definitely mutating it).

Reply:
I'm not too attached to it, but I would need to think of another name that felt good being called in a chain, like
```
.with_order(SortOrder::Ascending)
.with_start_rank(1)
.with_end_rank(3);
```
I suppose we could just do
```
.order(SortOrder::Ascending)
.start_rank(1)
.end_rank(3);
```
I also think it would be worth seeing if this style is more ergonomic for building config objects than the explicit builders we're currently using.

Reply:
yeah i was kind of thinking of just removing the `with_`, just like you did in your example there.

re: config builders, yeah i will have to dig into that more to form an opinion... in my head it boils down to whether the struct for the builder is the same struct for the final object or not. will keep it in mind though.

Reply:
will create a separate ticket for this